### ISSUE TITLE
without a ~/.gips/settings.py, GIPS is too noisy

### ISSUE THREAD

**Issue by justinfisk** [Applied-GeoSolutions/gips#189]
gips now seems to object if you do not have a user-level settings file

```Error loading '/home/justin/.gips/settings.py': [Errno 2] No such file or directory```

I believe it was introduced in this commit: a278382f12fcdf0fa4f2f48b7000792fb979e009

**Comment by bhbraswell** [comment]
If you copy gips/settings.py into ~/.gips/settings.py, then everything seems to work.
but that means you have so make settings.py without gips_config, because it also wants settings

**Comment by ircwaves** [comment]
Yup, noted this last night, but the hour was too late for me to be submiting bugs.  So I did what Rob suggests.  

@ags-tolson would you please revert your changes to `gips.utils.settings` so that user settings are not required.

**Comment by ags-tolson** [comment]
But they're not required?  The error is continuable?  It complains but it should keep going.

**Comment by justinfisk** [comment]
it complains a lot

**Comment by bhbraswell** [comment]
and fails to run. at least in my case

**Comment by ircwaves** [comment]
Also, my bad for not catching this on review.  The issue is that the first try is being used programatically:

1) Try to load user settings file
2) if that fails, load the system settings file

**Comment by ags-tolson** [comment]
@bhbraswell can you be more specific?  Maybe show us a gist of the output or something like that?

Regarding the noisiness though, can I suggest that going back to silent failures isn't the best solution?  Pointless noise is a legitimate concern, but so is wanting to be told when the config file you thought was there, isn't there.  What it ought to do I think is try to read the user settings file once, then if that fails, complain if verbosity is set to (say) 2 or higher.  But it should remember that it failed and not try again, relying on the global settings file for future calls to `settings()`.

I'm sorry about noisiness, I didn't realize how often `settings()` is called I guess.

**Comment by bhbraswell** [comment]
```
(.venv) braswell@sci3:~/gips$ gips_inventory modis -p indices -d 2002-001 -t h12v03 -v4 --fetch
Error loading '/home/braswell/.gips/settings.py': [Errno 2] No such file or directory
Error loading '/home/braswell/.gips/settings.py': [Errno 2] No such file or directory
Error loading '/home/braswell/.gips/settings.py': [Errno 2] No such file or directory
Error loading '/home/braswell/.gips/settings.py': [Errno 2] No such file or directory
Error loading '/home/braswell/.gips/settings.py':
Traceback (most recent call last):
  File "/home/braswell/gips/gips/utils.py", line 418, in cli_error_handler
    yield
  File "/home/braswell/gips/gips/utils.py", line 154, in settings
    src = imp.load_source('settings', settings_path)
IOError: [Errno 2] No such file or directory
Error loading '/home/braswell/.gips/settings.py':
Traceback (most recent call last):
  File "/home/braswell/gips/gips/utils.py", line 418, in cli_error_handler
    yield
  File "/home/braswell/gips/gips/utils.py", line 154, in settings
    src = imp.load_source('settings', settings_path)
IOError: [Errno 2] No such file or directory
GIPS Data Inventory (v0.9.0-dev)
Retrieving inventory for site tiles
Error loading '/home/braswell/.gips/settings.py':
Traceback (most recent call last):
  File "/home/braswell/gips/gips/utils.py", line 418, in cli_error_handler
    yield
  File "/home/braswell/gips/gips/utils.py", line 154, in settings
    src = imp.load_source('settings', settings_path)
IOError: [Errno 2] No such file or directory
Error loading '/home/braswell/.gips/settings.py':
Traceback (most recent call last):
  File "/home/braswell/gips/gips/utils.py", line 418, in cli_error_handler
    yield
  File "/home/braswell/gips/gips/utils.py", line 154, in settings
    src = imp.load_source('settings', settings_path)
IOError: [Errno 2] No such file or directory
Error loading '/home/braswell/.gips/settings.py':
Traceback (most recent call last):
  File "/home/braswell/gips/gips/utils.py", line 418, in cli_error_handler
    yield
  File "/home/braswell/gips/gips/utils.py", line 154, in settings
    src = imp.load_source('settings', settings_path)
IOError: [Errno 2] No such file or directory
Retrieved MCD43A4.A2002001.h12v03.006.2016123182450.hdf
Error loading '/home/braswell/.gips/settings.py':
Traceback (most recent call last):
  File "/home/braswell/gips/gips/utils.py", line 418, in cli_error_handler
    yield
  File "/home/braswell/gips/gips/utils.py", line 154, in settings
    src = imp.load_source('settings', settings_path)
IOError: [Errno 2] No such file or directory
Error loading '/home/braswell/.gips/settings.py':
Traceback (most recent call last):
  File "/home/braswell/gips/gips/utils.py", line 418, in cli_error_handler
    yield
  File "/home/braswell/gips/gips/utils.py", line 154, in settings
    src = imp.load_source('settings', settings_path)
IOError: [Errno 2] No such file or directory
MCD43A4.A2002001.h12v03.006.2016123182450.hdf -> /data/gips/modis/tiles/h12v03/2002001/MCD43A4.A2002001.h12v03.006.2016123182450.hdf
1 files (1 links) from /data/gips/modis/stage added to archive in 0:00:00.016565
Error loading '/home/braswell/.gips/settings.py':
Traceback (most recent call last):
  File "/home/braswell/gips/gips/utils.py", line 418, in cli_error_handler
    yield
  File "/home/braswell/gips/gips/utils.py", line 154, in settings
    src = imp.load_source('settings', settings_path)
IOError: [Errno 2] No such file or directory

Asset Holdings for tile h12v03
    DATE     MCD12Q1   MCD43A2   MCD43A4   MOD09Q1   MOD10A1   MOD10A2   MOD11A1   MOD11A2   MYD10A1   MYD10A2   MYD11A1   MYD11A2   Product  
2002        
    001                           100.0%                                                                                             


1 files on 1 dates

SENSORS
MCD: Aqua/Terra Combined
MOD: Terra
MOD-MYD: Aqua/Terra together
MYD: Aqua
Fatal: 8 error(s) occurred:
Error loading '/home/braswell/.gips/settings.py':
None
Error loading '/home/braswell/.gips/settings.py':
None
Error loading '/home/braswell/.gips/settings.py':
None
Error loading '/home/braswell/.gips/settings.py':
None
Error loading '/home/braswell/.gips/settings.py':
None
Error loading '/home/braswell/.gips/settings.py':
None
Error loading '/home/braswell/.gips/settings.py':
None
Error loading '/home/braswell/.gips/settings.py':
None
```

**Comment by bhbraswell** [comment]
My apologies, I guess it is working. I previously just saw what looked like real errors and quit.

**Comment by ircwaves** [comment]
the absence of user settings is not a problem, and should not be notified.

**Comment by ircwaves** [comment]
I will document the long-standing issue that settings need to be re-worked, but for now the changes to `gips.utils.settings` should be reverted.

**Comment by ircwaves** [comment]
instead of reverting, there could be a check for the existence of the file before attempting to load it.

**Comment by ags-tolson** [comment]
So I think I have a fix, in PR #191.

**Comment by ags-tolson** [comment]
this merged months ago, closing
